src/popover/container.scss

```diff
@@ -6,12 +6,60 @@
 @use '../internal/styles' as styles;
 @use '../internal/styles/tokens' as awsui;
 
+$arrow-height: 10px;
+
 .container {
   display: inline-block;
   position: fixed;
   inset-block-start: -9999px;
   inset-inline-start: -9999px;
   z-index: 2000;
+
+  // A little pseudoelement to extend the container to the space between the
+  // popover and the trigger to avoid hover-based popovers/tooltips from
+  // closing too eagerly when the cursor goes between them.
+  &::before {
+    content: '';
+    position: absolute;
+  }
+
+  &:has(&-arrow-position-bottom-left),
+  &:has(&-arrow-position-bottom-center),
+  &:has(&-arrow-position-bottom-right) {
+    &::before {
+      inset-inline: 0;
+      inset-block-start: -$arrow-height;
+      block-size: $arrow-height;
+    }
+  }
+
+  &:has(&-arrow-position-top-left),
+  &:has(&-arrow-position-top-center),
+  &:has(&-arrow-position-top-right) {
+    &::before {
+      inset-inline: 0;
+      inset-block-end: -$arrow-height;
+      block-size: $arrow-height;
+    }
+  }
+
+  &:has(&-arrow-position-right-top),
+  &:has(&-arrow-position-right-bottom) {
+    &::before {
+      inset-block: 0;
+      inset-inline-start: -$arrow-height;
+      block-size: $arrow-height;
```

Comment:
Looks like `block-size` should be `inline-size` in this and the next rule set.

I've added some styles with the dev tools to debug this pseudoelement, and it only has the expected position and dimensions when the popover/tooltip is anchored to the top or bottom, not left or right.

![Screenshot 2025-03-26 at 08 24 16](https://github.com/user-attachments/assets/05e1e9f0-a761-4793-879f-ba1515bec284)
![Screenshot 2025-03-26 at 08 24 57](https://github.com/user-attachments/assets/85803d3a-2e39-4979-bbe8-0c5106057420)